File: Test10/equi_filter.pl
#!/usr/bin/perl -w

foreach $line(<STDIN>){
	@words = grep(/./, split(" ", $line));
	undef @result;
	foreach $thing (@words){
		undef %data;
		$buffer = $thing;
		while($thing =~ /\w/){
			$thing =~ /^(\S)(.*)$/;
			if(!$data{lc $1}){
				$data{lc $1} = 1;
			}else{
				$data{lc $1}++;
			}
			$thing = $2;
		}
		$last = -1;
		foreach $element(keys %data){
			$equalwith = 1;
			if($last == -1){
				$last = $data{$element};
				next;
			}
			if($last != $data{$element}){
				$equalwith = 0;
				last;
			}
		}
		push @result, $buffer if($equalwith);
	}
	print "@result\n";
}

File: Test10/equi_filter.py
#!/usr/bin/python3

import sys
line = sys.stdin.readline()
while line:
	words = line.split(' ')
	result = []
	for word in words:
		word = word.strip()
		temp = word
		word = word.lower()
		data = {}
		for letter in word:
			if letter not in data: data[letter] = 1
			else: data[letter] += 1
		last = -1
		equalwith = 1
		for element in data:
			if last == -1: last = data[element]
			else:
				if data[element] != last:
					equalwith = 0
					break
		if equalwith == 1: result.append(temp)
	output = ' '.join(result)
	print(output)
	line = sys.stdin.readline()	
